src/ch05-02-example-structs.md

```diff
@@ -158,62 +167,70 @@ struct Rectangle {
-for direct end user consumption. The primitive types we’ve seen so far
-implement `Display` by default, because there’s only one way you’d want to show
-a `1` or any other primitive type to a user. But with structs, the way
-`println!` should format the output is less clear because there are more
-display possibilities: Do you want commas or not? Do you want to print the
-curly brackets? Should all the fields be shown? Due to this ambiguity, Rust
-doesn’t try to guess what we want, and structs don’t have a provided
-implementation of `Display`.
-
-If we continue reading the errors, we’ll find this helpful note:
+Makro `println!` może formatować na wiele sposobów, a domyślnie para nawiasów klamrowych
+daje `println!` znać, że chcemy wykorzystać formatowanie `Display` (ang. wyświetlenie).
+Jest to tekst przeznaczony dla docelowego użytkownika.
+Widziane przez nas wcześniej prymitywne typy implementują `Display` automatycznie,
+bo przecież jest tylko jeden sposób wyświetlenia użytkownikowi symbolu `1` czy 
+jakiegokolwiek innego prymitywnego typu.
+Ale kiedy w grę wchodzą struktury, sposób w jaki `println!` powinno formatować
+tekst jest mniej oczywiste, bo wyświetlać strukturę można na wiele sposobów:
+z przecinkami, czy bez?
+Chcesz wyświetlić nawiasy klamrowe?
+Czy każdy atrybut powinien być wyświetlony?
+Przez tę wieloznaczność Rust nie zakłada z góry co jest dla nas najlepsze, 
+więc z tego powodu struktury nie implementują automatycznie cechy `Display`.
+
+Jeśli będziemy czytać dalej znajdziemy taką przydatną informację:
 
 ```text
 `Rectangle` cannot be formatted with the default formatter; try using
 `:?` instead if you are using a format string
 ```
 
-Let’s try it! The `println!` macro call will now look like `println!("rect1 is
-{:?}", rect1);`. Putting the specifier `:?` inside the curly brackets tells
-`println!` we want to use an output format called `Debug`. The `Debug` trait
-enables us to print our struct in a way that is useful for developers so we can
-see its value while we’re debugging our code.
+Jesteśmy poinformowani, że podana przez nas struktura nie może być użyta
+z domyślnym formaterem i zasugerowane jest nam użycie specyfikatora formatowania `:?`.
+To tak też zróbmy! Wywołanie makra `println!` teraz będzie wyglądać następująco:
+`println!("rect1 to {:?}", rect1);`. 
+Wprowadzenie specyfikatora `:?` wewnątrz pary nawiasów klamrowych
+przekazuje `println!`, że chcemy użyć formatu wyjścia o nazwie `Debug`.
+Cecha `Debug` pozwala nam wypisać strukturę w sposób użyteczny dla deweloperów,
+czyli ułatwia nam zajrzeć do wartości wewnątrz struktury podczas debugowania przez nas kodu.
 
-Run the code with this change. Drat! We still get an error:
+Uruchom kod z tymi zmianami. A niech to! Nadal pojawia się komunikat o błędzie:
 
 ```text
 error[E0277]: the trait bound `Rectangle: std::fmt::Debug` is not satisfied
 ```
 
-But again, the compiler gives us a helpful note:
+Ale kompilator nas nie opuszcza:
```

Comment:
Ojoj, to już kolokwializm. Proponuję:
Ale kompilator ponownie daje nam pomocny komunikat:

Reply:
Chciałem przekazać przyjazność i pomocność kompilatora, ale rozumiem. Zmienię.